enumerables.rb

```diff
@@ -0,0 +1,269 @@
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    arr = is_a?(Array) ? self : to_a
+    is_a?(Hash) ? arr.length.times { |i| yield(arr[i][0], arr[i][1]) } : arr.length.times { |i| yield(arr[i]) }
+    self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    arr = to_a
+    (0...arr.length).my_each do |i|
+      yield(arr[i], i)
+    end
+    self
+  end
+
+  def my_map(&proc)
+    my_arr = []
+    hash = {}
+    if block_given?
+      is_a?(Hash) ? my_each { |k, v| hash[k] = yield(k, v) } : my_each { |x| my_arr << yield(x) }
+    else
+      my_each { |x| proc.call(x) }
+    end
+    is_a?(Hash) ? hash : my_arr
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    if is_a?(Array)
+      my_arr = []
+      my_each do |i|
+        my_arr << i if yield(i)
+      end
+      my_arr
+    elsif is_a?(Hash)
+      my_hash = {}
+      my_each do |i, j|
+        my_hash.store(i, j)
+      end
+    end
+  end
+
+  def my_all?(*args)
+    is_true = true
+    if block_given?
+      if is_a?(Hash)
+        to_a.my_each do |k, v|
+          is_true = false unless yield(k, v)
+        end
+      else my_each { |x| is_true = false unless yield(x) }
+      end
+    elsif args[0].is_a?(Integer)
+      my_each { |x| is_true = false unless args[0] == x }
+    elsif args[0].nil?
+      my_each { |x| is_true = false if x == false || x.nil? }
+    elsif args[0].is_a?(Regexp)
+      my_each { |x| is_true = false if x.match(args[0]).nil? }
+    else
+      my_each { |x| is_true = false unless x.is_a?(args[0]) }
+    end
+    is_true
+  end
+
+  def my_count(num = 0)
+    sum = 0
+    length.times { |x| sum += 1 if self[x] == num } if num != 0
+    sum = max if !block_given? && num.zero?
+    if block_given?
+      my_each do |i|
+        sum += 1 if yield(i)
+      end
+    end
+    sum
+  end
+
+  def my_none?(obj = nil)
+    none = true
+    if block_given?
+      my_each { |i| none = false if yield(i) }
+
+    elsif obj.nil?
+      my_each { |i| none = false if i }
+    elsif obj.is_a?(Regexp)
+      my_each { |i| none = false if i =~ obj }
+    else
+      my_each { |i| none = false if i.is_a?(obj) }
+    end
+
+    none
+  end
+
+  def my_any?(*args)
+    any = false
+    if block_given?
+      my_each { |x| any = true if yield(x) }
+    elsif args[0].is_a?(Integer)
+      my_each { |x| any = true if x == args[0] }
+    elsif args[0].nil?
+      my_each { |x| any = true if x }
+    elsif args[0].is_a?(Regexp)
+      my_each { |x| any = true if x =~ args[0] }
+    else
+      my_each { |x| any = true if x.is_a?(args[0]) }
+    end
+    any
+  end
+
+  # def my_inject(i = 0)
+  #   i = to_a[0].is_a?(String) ? to_a[0] : i
+  #   my_each do |j|
+  #     i = yield(i,j)
+  #   end
+  #   i
+  # end
+
+  def my_inject(*args)
```

Comment:
Consider:
- When a symbol is specified without an initial value
- when a symbol is specified with the initial value.

Basically consider when argument  1 is a symbol and of course if there is a second argument the second argument is a symbol